i18n/changed.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+"""
+Determine if the source translation files are up-to-date.
+"""
+from __future__ import print_function
+from subprocess import CalledProcessError
+
+from i18n import Runner
+from i18n.execute import execute
+
+
+class Changed(Runner):
+    """
+    Class used to check if the source translation files are up-to-date
+    """
+    def run(self, args):
+        """
+        Main entry point of script
+        """
+        changes_detected = self.detect_changes()
+        message = self.get_message(changes_detected)
+        print(message)
+        if changes_detected:
+            return 1
+        return 0
+
+    def detect_changes(self):
+        """
+        Detect if changes have been made to the msgid or msgstr lines in the translation files.
+
+        Returns:
+          boolean: True, if changes detected; otherwise, False.
+
+        Note:
+          This method requires ``git`` to be installed on the executing machine.
+        """
+        try:
+            execute('git diff --exit-code -G "^[msgid|msgstr]"')
```

Comment:
The regex here is too greedy, and will match any line beginning with any of the characters in the words `msgid` or `msgstr`. It should be updated to `^(msgid|msgstr)`.

A test should be added for this, too. I detected this when I updated `local.txt` to install this package from Git instead of PyPI:

```diff
-edx-i18n-tools==0.3.3
+git+https://github.com/edx/i18n-tools.git@iivic/SOL-2053#egg=i18n_tools==0.3.3
```